toolchain/lower/mangler.cpp

```diff
@@ -0,0 +1,91 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/lower/mangler.h"
+
+#include "toolchain/base/kind_switch.h"
+#include "toolchain/sem_ir/entry_point.h"
+
+namespace Carbon::Lower {
+
+auto Mangler::MangleInverseQualifiedNameScope(bool first_name_component,
+                                              llvm::raw_ostream& os,
+                                              SemIR::NameScopeId name_scope_id)
+    -> void {
+  auto& sem_ir = file_context_.sem_ir();
+  auto names = sem_ir.names();
```

Comment:
Think I got those addressed.

I am not using vscode at the moment, but really should give it a go - and running it with bazel is a bit tricky since the project isn't clalng-tidy clean at the moment.

Reply:
I believe the project is clang-tidy clean under LLVM 16 -- we are running it on every commit at that version. If you're having issues, fixes are welcome, but also using LLVM 16 should be clean.

Reply:
ah, yep - didn't realize the bazel build used my host tools like that - I'd assumed it was more hermetic. got my local situation version matched so that'll be easier (until I forget to update my PATHs after a reboot between LLVM and Carbon development... )